<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.0//EN">
<html>
<head>
<meta name="copyright" content="2003,  Bradley Kjell" />
<meta name="author" content="Bradley Kjell kjell@ieee.org " />
<meta name="translation" content="Copyright 2004 Deutsche &Uuml;bersetzung Heinrich Gailer heinrich@gailer-net.de" />
<meta name="robots" content="index,follow" />
<title>Exception-Objekte</title>
<link rel="stylesheet" type="text/css" href="../CAIstyle.css" />
</head>

<body>
<!-- generated by CSSmaker.java, version of 10/12/2004-->
<!-- from source javaChap81.frames on Wed Oct 24 11:17:05 CEST 2007 -->

<!-- ANSWER AREA -->

<div class="answer">
<h3>Eine gute Antwort w&auml;re:</h3>
<p>
<code>exp</code> verweist auf ein <code>NumberFormatException</code>-Objekt.
</p>


</div>

<!-- LESSON AREA -->
<div class="lesson">
<h1><code><strong>Exception</strong></code>-Objekte</h1>

<p>
Hier ist eine try/catch-Struktur:
</p>

<pre class="program">
try
{
  // Anweisungen, von denen einige eine Ausnahme ausl&ouml;sen k&ouml;nnten
}

catch ( <em>IrgendeinExceptionTyp</em> ex )
{
  // Anweisungen um diesen Ausnahmetyp zu behandeln
}

....  // m&ouml;glicherweise weitere catch-Blocks

....  // m&ouml;glicherweise ein finally-Block

</pre>


<A NAME="exception object"></A>
<p>
Wenn ein <code>catch{}</code>-Block die Kontrolle bekommt (mit der Ausf&uuml;hrung beginnt),
besitzt er eine Referenz auf ein Objekt der Klasse <code>Exception</code>
(oder auf eine Subklasse von <code>Exception</code>).
Die spezifische Klasse des Objekts h&auml;ngt davon ab was f&uuml;r eine Ausnahme ausgel&ouml;st wurde.
</p>

<A NAME="Exception.printStackTrace()"></A>
<A NAME="Exception.getMessage()"></A>

<p>
Wenn w&auml;hrend der Programmausf&uuml;hrung ein Exception-Ereignis auftritt,
&uuml;bernimmt das Java Laufzeitsystem und
erzeugt ein <code>Exception</code>-Objekt, um dieses Ereignis zu repr&auml;sentieren.
Die Information &uuml;ber das Ereignis wird in das Objekt gestellt.
Wenn die Ausnahme innerhalb eines <code>try{}</code>-Blocks auftrat,
schickt das Java Laufzeitsystem das
<code>Exception</code>-Objekt an den geeigneten
<code>catch{}</code>-Block (wenn es einen gibt).

<p>
<code>Exception</code>-Objekte sind Java Objekte.
Sie verf&uuml;gen &uuml;ber Instanzvariablen und Instanzmethoden darunter:
</p>

<ul>
<li><code>public void <b>printStackTrace()</code></b>
    <ul>
    <li>Ein Stack-Protokoll ausgeben &#8213; eine Liste,
        die eine Sequenz von Methodenaufrufen bis zu dieser Ausnahme zeigt.
        </li></ul></li>
<li><code>public String <b>getMessage()</code></b>
    <ul>
    <li>Gibt einen String zur&uuml;ck,
        der beschreiben k&ouml;nnte was schief gegangen ist.</li></ul>
</ul>

<p>
Ein <code>catch{}</code>-Block kann diese Methoden verwenden
um eine informative Fehlermeldung auf den Bildschirm auszugeben.
</p>



</div>

<!-- QUESTION AREA -->
<div class="question">
<h3> FRAGE 2:</h3>
<a href="ch81_3.html">
<img src="../nextBut.gif" hspace="5" align="right" border="0"/></a>
<p>
Angenommen ein Array wurde deklariert durch:
</p>

<pre class="code">
int[] wert = new int[10];
</pre>

<p>
Ist es zul&auml;ssig sich auf <code>wert[10]</code> zu beziehen?
</p>

</div>
<div>
<a title="Zurück zum Inhaltsverzeichnis gehen." href="../../java-toc.html#81">Inhaltsverzeichnis</a>
</div>

</body>
</html>
